skyportal/handlers/api/source.py

```diff
@@ -1899,6 +1899,49 @@ def post_source(data, user_id, session, refresh_source=True):
             "of group_ids for which if there is a source, cancel saving to the associated group_id from the key"
         )
 
+    # we want to allow admins to save sources as another user(s).
+    # it's optional, and we default to saving to the current user unless specified otherwise.
+    saver_per_group_id = {gid: user for gid in group_ids}
+    if 'saver_per_group_id' in data:
+        if not user.is_admin:
+            raise AttributeError(
+                "You must be an admin to specify a saver_per_group_id field."
+            )
+        if not isinstance(data['saver_per_group_id'], dict):
+            raise AttributeError(
+                "Invalid saver_per_group_id field. Please specify a dict"
+            )
+        try:
+            saver_id_per_group_id = {
+                int(gid): int(user_id)
+                for gid, user_id in data['saver_per_group_id'].items()
+            }
+        except Exception:
+            raise AttributeError(
+                "Invalid saver_per_group_id field. Please specify a dict with group_ids as keys and user_ids as values"
+            )
+
+        try:
+            for gid in saver_id_per_group_id:
+                if gid in group_ids:
+                    group_saver_for_gid = session.scalar(
+                        sa.select(GroupUser).where(
+                            GroupUser.user_id == saver_id_per_group_id[gid],
+                            GroupUser.group_id == gid,
+                        )
+                    )
+                    if not group_saver_for_gid:
+                        log(
+                            f"Could not save to group {gid} as user {saver_id_per_group_id[gid]} (user is not a member of the group). Using current user {user.id} instead."
```

Comment:
Too risky (as mentioned in the PR description). If for some reason the account specified to save the source is removed from the group, Kowalski would end up not saving the object to any group.

Seems like an unnecessary risk. Maybe I can return it as a warning (or a list of warnings) to the client though?

Reply:
Sounds good. Just would be good to somehow flag it for folks. I just worry sometimes about logs that don't get read :)

Reply:
Yes I agree.  But anyway this is a super admin only thing that will only be used by Kowalski, so I'm not too worried about that. 

I'll figure something out to have the `post_source` method return a list of warnings if any, so that we can return those to the client.

Reply:
@mcoughlin I will return the warnings anyway, but what I can do is skip saving to these groups with incorrect users instead of using that fallback (and not erroring out). What do you think is best?

Reply:
Sounds good to me :)